### ISSUE TITLE
GraphQL returning null with `hasMany` relationship when query has more than 3 levels nested

### ISSUE THREAD

**Issue by willianjusten** [strapi/strapi#8396]
## Bug report

If we have a `hasMany` relationship with nested components with more than 3 levels, the GraphQL just returns `null` instead of the correct data.

I analyzed all the versions and found out that the bug started here: https://github.com/strapi/strapi/pull/7959 , which was released on https://github.com/strapi/strapi/releases/tag/v3.1.6

### Steps to reproduce the behavior

1. Clone this [repo](https://github.com/willianjusten/strapi-nested-bug-graphql)
2. Set it up with `yarn` and run it with `yarn develop`
3. Go to the admin panel at `http://localhost:1337/admin`
4. Create an `Author` with at least one `SocialNetwork`.
5. Create a `LandingPage` and add the created author
6. Go to `Settings > User & Permissions > Roles` and enable `find` to `LandingPage` and `Authors`
7. Test the following queries

This one will return `null`:

```graphql
{
  landingPage {
    SectionAboutUs {
      authors {
        SocialNetworks {
          Network
          url
        }
      }
    }
  }
}
```

This one will return the correct data:

```graphql
{
  authors {
    SocialNetworks {
      Network
      url
    }
  }
}
```

### Expected behavior

It should return the data even when it has more than 3 nested levels. If you test the version `3.1.5` and below or just remove the [line added here](https://github.com/strapi/strapi/pull/7959/files#diff-dbd596fb02f5c66a01a212d650e2d62644303a30a132a8ace1370c08e89fbe05R147) it will show the correct behavior.

### Screenshots

More than 3 levels nested, getting `null` as value:

![image](https://user-images.githubusercontent.com/3991845/96530848-af2b2c00-125e-11eb-9c6f-be7f75647acf.png)

Same data now returned correctly when less nested query:

![image](https://user-images.githubusercontent.com/3991845/96530900-cc5ffa80-125e-11eb-830c-d74b7fa2278c.png)

### System

- Node.js version: v12.19.0
- NPM version: 6.14.8
- Strapi version: 3.2.4
- Database: SQLite / Postgres (tested on both and had the same error)
- Operating system: Ubuntu 20.04 and Windows 10 WLS2

**Comment by derrickmehaffy** [comment]
Some things to note:

- I was able to reproduce this on v3.2.4
- Attempting to use the REST `localhost:1337/landing-page` yields a 404 error
- The above leads me to believe it may not just be a GraphQL error but a controller one :thinking: 

@alexandrebodin do we have different controllers for single-types? Overriding the landing-page controller with one of our core-controller yields an error.

**Comment by alexandrebodin** [comment]
Hello we are aware of this bug and working on it but reverting the introduced bug isn't a good fix either so we are digging more into it right now :)

**Comment by willianjusten** [comment]
Thanks for the quick response @alexandrebodin , I hope you find a good solution for this soon =)
I'll see if I get some time to dig on it and maybe provide a solution, but probably you will be faster than me to understand the code.

**Comment by mihaiblaga89** [comment]
stumbled upon the same issue. Weird thing is, it only happens with mongodb as a db. Locally with sqlite it works just fine

**Comment by DanielHoffmann** [comment]
@mihaiblaga89 I ran into this bug in v3.3.3 using postgres as well

**Comment by frvasquezjaquez** [comment]
Same issue with mongo, i had to downgrade to  3.1.5

**Comment by ramonmulia** [comment]
it is happening with postgres strapi@3.3.3 (docker images)

**Comment by alexandrebodin** [comment]
Going to close as this was fixed in recent releases 3.5.x
